### ISSUE TITLE
Recompiling mess

### ISSUE THREAD

**Issue by hlsantos** [jstedfast/MailKit#1036]
Hello,  I want to say I am not a complainer.  Honestly.  I always roll up my sleeves to try solve issue and problems  on my own that are not under my control before going to developers and posting an issue.  But I just lost 4 hours of my life trying to recompile MailKit and MimeKit.    This work was is in reference to  https://github.com/win-acme/win-acme/issues/1567 and the suggestion to use MailKit in lieu of .Net SmtpClient.  So I am trying to build Mailkit and explore this solution for win-acme email notifications.

 I am using the following environment.

- VS2015
- Windows 2012R

I followed the instructions in the README.md to git clone MailKit and also the git submodules  for MimeKit.  

 I am getting all sorts of compilers errors.   I figure it was related to the missing Bouncy assembly. I noticed the reference assembly was missing so  I use Nuget to install it, but the reference was still moving.  However, with the added Nuget installed assembly (world icon symbol),  the "using Org.Bouncy.xxxxx" assemblies were resolved.  Yet, I am getting many compiler errors for number of many things. I believe it starts with missing DKIM Arc related stuff.  But it was too much for me to continue with this, probably best to ask the developer(s).  See the compiler output below.

I apologize for my post. Its not really a bug per se, maybe just a doc issue., but this should be more straight forward.    I am going to suggest in another new issue post regarding DKIM and ARC a COMPILER option.  I will explain more in my Suggestion issue post.

Thanks

Output from compile:

1>------ Build started: Project: MimeKit.Net45, Configuration: Release Any CPU ------
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcSigner.cs(346,66,346,67): error CS0443: Syntax error; value expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Tnef\TnefPart.cs(458,90,458,96): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcSigner.cs(346,68,346,72): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcSigner.cs(346,78,346,81): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcSigner.cs(346,82,346,87): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(291,41,291,47): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(291,48,291,49): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(315,33,315,36): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(315,36,315,37): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(315,36,315,37): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(315,38,315,40): error CS1525: Invalid expression term '&&'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(315,84,315,85): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(315,84,315,85): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(369,33,369,36): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(369,36,369,37): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(369,36,369,37): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(369,38,369,40): error CS1525: Invalid expression term '&&'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(369,84,369,85): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(369,84,369,85): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(416,86,416,93): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Tnef\TnefPart.cs(458,97,458,102): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(520,55,520,61): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(520,62,520,64): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Tnef\TnefPart.cs(463,70,463,81): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(548,63,548,64): error CS0443: Syntax error; value expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(548,65,548,69): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(548,75,548,78): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\ArcVerifier.cs(548,79,548,84): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(61,61,61,64): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(61,74,61,77): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(71,61,71,64): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(71,74,71,77): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(99,61,99,64): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(99,74,99,77): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(109,61,109,64): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(109,74,109,77): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(178,40,178,46): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Utils\MimeUtils.cs(160,82,160,88): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Utils\MimeUtils.cs(160,89,160,94): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(178,46,178,47): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(178,46,178,47): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(181,40,181,46): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(181,46,181,47): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(181,46,181,47): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(224,40,224,46): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(224,46,224,47): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(224,46,224,47): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(227,40,227,46): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(227,46,227,47): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(227,46,227,47): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(366,43,366,56): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(366,56,366,57): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(366,56,366,57): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(370,40,370,53): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(370,53,370,54): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(370,53,370,54): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(374,33,374,45): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(374,45,374,46): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(374,45,374,46): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(378,39,378,51): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(378,51,378,52): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(378,51,378,52): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(406,50,406,63): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(406,63,406,64): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(406,63,406,64): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(410,47,410,60): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(410,60,410,61): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AsymmetricAlgorithmExtensions.cs(410,60,410,61): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AuthenticationResults.cs(432,68,432,71): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AuthenticationResults.cs(432,72,432,79): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AuthenticationResults.cs(759,69,759,72): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AuthenticationResults.cs(759,73,759,74): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Utils\ParseUtils.cs(478,90,478,96): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Utils\ParseUtils.cs(478,97,478,103): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AuthenticationResults.cs(808,67,808,70): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\AuthenticationResults.cs(808,71,808,78): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Header.cs(714,75,714,82): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimSignerBase.cs(173,46,173,50): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimSignerBase.cs(173,50,173,51): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimSignerBase.cs(173,50,173,51): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimSignerBase.cs(175,7,175,11): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(71,42,71,48): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(71,49,71,50): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(89,41,89,47): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(89,48,89,50): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(136,33,136,36): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(261,81,261,87): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(261,88,261,96): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(261,102,261,105): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(261,106,261,113): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(136,36,136,37): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(136,36,136,37): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(136,38,136,40): error CS1525: Invalid expression term '&&'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(136,84,136,85): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifier.cs(136,84,136,85): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(344,85,344,91): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(344,92,344,99): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(344,105,344,108): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(344,109,344,111): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(390,85,390,91): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(390,92,390,99): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(212,42,212,48): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(212,49,212,50): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(243,41,243,47): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(243,48,243,49): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(244,83,244,86): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(244,87,244,96): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(254,41,254,47): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(254,48,254,49): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(261,41,261,47): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(261,48,261,49): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(287,42,287,48): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\DkimVerifierBase.cs(287,49,287,50): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\OpenPgpContext.cs(767,40,767,46): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\OpenPgpContext.cs(767,92,767,93): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\OpenPgpContext.cs(767,92,767,93): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Cryptography\OpenPgpContext.cs(780,51,780,56): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(390,105,390,108): error CS1525: Invalid expression term 'int'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MailboxAddress.cs(390,109,390,111): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MessagePart.cs(86,28,86,32): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MessagePart.cs(86,32,86,33): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MessagePart.cs(86,32,86,33): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(140,22,140,28): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(140,28,140,29): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(140,28,140,29): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(145,35,145,42): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(145,42,145,43): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(145,42,145,43): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(293,88,293,94): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(293,95,293,97): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(1070,106,1070,112): error CS1525: Invalid expression term 'string'
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeEntity.cs(1070,113,1070,118): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(105,25,105,28): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(105,28,105,29): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(105,28,105,29): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(113,23,113,26): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(113,26,113,27): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(113,26,113,27): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextPart.cs(325,34,325,42): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextRfc822Headers.cs(75,28,75,32): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextRfc822Headers.cs(75,32,75,33): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\TextRfc822Headers.cs(75,32,75,33): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeParser.cs(681,68,681,74): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimePart.cs(108,29,108,31): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimePart.cs(108,31,108,32): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimePart.cs(108,31,108,32): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimePart.cs(116,23,116,29): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimePart.cs(116,29,116,30): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimePart.cs(116,29,116,30): error CS1513: } expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\MimeTypes.cs(1008,46,1008,54): error CS1003: Syntax error, ',' expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Multipart.cs(109,27,109,33): error CS1026: ) expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Multipart.cs(109,33,109,34): error CS1002: ; expected
1>C:\local\MailKit\submodules\MimeKit\MimeKit\Multipart.cs(109,33,109,34): error CS1513: } expected
2>------ Build started: Project: MailKit.Net45, Configuration: Release Any CPU ------
2>CSC : error CS0006: Metadata file 'C:\local\MailKit\submodules\MimeKit\MimeKit\bin\Release\net45\MimeKit.dll' could not be found
========== Build: 0 succeeded, 2 failed, 0 up-to-date, 0 skipped ==========

**Comment by jstedfast** [comment]
I suspect that the problem is that VS2015 doesn't support the newer C# syntax versions that MimeKit makes use of. I honestly haven't tried to build MimeKit with anything older than VS2017 and am currently using VS2019 to do all of my development (up until a few weeks ago, I used VS for Mac and VS2017 on Windows to make releases).

My guess is that this isn't possible to build using VS2015 unless there's a way to upgrade the C# compiler.

**Comment by hlsantos** [comment]
Hi, This appears to be ARC related so maybe my suggestion would help. Maybe I can try to port it back to "VS2015" support?  

I do have VS2019, but it has number of security and privacy (namely telemetry concerns) issues I need to get worked out before the switch of build and production is done.

**Comment by jstedfast** [comment]
From what I remember when I looked at the build errors you got yesterday, it looked like the problem was syntax like this:

```csharp
if (entity is Multipart multipart) {
    ...
```

This is a newer (more convenient) C# syntax that extends the usage of the `is` keyword. I think it was introduced in C# 7 (called "pattern matching"): https://docs.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-7#pattern-matching

**Comment by jstedfast** [comment]
FWIW, it looks like there might be a way for you to get C# 7 working in VS2015: https://stackoverflow.com/questions/39461407/how-to-use-c-sharp-7-with-visual-studio-2015

I would rather not revert to older C# syntax if it can be avoided because it's more work for me to maintain.

**Comment by jstedfast** [comment]
Is there a reason that you are building from source instead of using pre-built nugets? Using pre-built nugets seems to be the standard.

**Comment by hlsantos** [comment]
Hi, in all honestly, this is really my first adventure with "open source" git/hub collaboration.  I am an old cvs guy, well, not that old. <g>.  But I have assisted authors and rather than get involved with process, i.e. "step on toes,"  I pass on my work - changes, diffs, suggestion, fixes, etc..   

In the last 3 months, during this crazy pandemic "Stay Home" period, I have been moving my platform to GitHub. Except for the SDK (https://github.com/hlsantos/wcSDK), everything else is private until its all organized and prepared for controlled Git/Hub collaboration with 3rd party developers.  I am far from being finished with this humongous project.

In short, tell me what I need to do.  I have never used Nuget too.  Its seem to be a dependent library packaging tool?

I was able to recompile by using "normal" backward compatible syntax.  The changes are 99% related to the new pattern matching syntax that helps to reduce type casting and the 1% related to the two spots using the new Discard syntax, i.e. using out _ as a parameter.

Thanks

**Comment by hlsantos** [comment]
> FWIW, it looks like there might be a way for you to get C# 7 working in VS2015: https://stackoverflow.com/questions/39461407/how-to-use-c-sharp-7-with-visual-studio-2015
> 
> I would rather not revert to older C# syntax if it can be avoided because it's more work for me to maintain.

I see your point. I have learned over 20+ languages since the late 70s. Ok, I am old. <g> I am have attributed my consumption starting with the APL language, perhaps the most cryptic language, so cryptic it had its own keyboard:  https://www.google.com/search?q=APL+Keyboard

I personally believe APL found its home with Deep AI.  But from APL everything I come to feel being specific is terrific and code with for long term readability  in mind.   I believe the C# features reduces coding, but for this particular thing, its sort of has a syntaxform that is not nature for how variables are in scope.  Its not like the new syntax offer a new form of computing power like an APL function using one to two symbols  to replace 5, 10, 20 lines of code.    This is more about reducing type casting. I personally like to type case because its more readable.  Now the Discard syntax. is an interesting new language feature because it can help with memory optimization.  I sort of like it, but its not compatible.   I believe you can do the same thing using overloads 

Anyway, I do see your point.  But for these two things? Its a language syntax thing that really does not provide any benefit but reducing your typing.  I would side with compatibility.

Of course, the above is just my opinion.

Thanks

**Comment by hlsantos** [comment]
Hi Jeffrey,  I am going to close this issue.   The end result is that this wass a C# 7.0 (8.0?) compiler version issue and there is a solution using the Nuget package for Microsoft.Net.Compilers.  

While I did successfully recompiled MailKit and MimeKit by making the minor changes where newer C# compiler pattern matching and discard syntax was used,  I do agree if there a way to provide a provisional update to VS2015 compiler (C# Compiler version 1.3.1.60616), then there is no need to change the code.  Maybe include a "Build-Requirements.md" with notes about using VS2015.   I am going to first read up on this "Nuget" stuff before I attempt to change my VS2015 environment.   Maybe I can use it for my migration work to GitHub.  I already learned a lot from your setup. :-) 

Thanks for your patience and assistance.
